In [18]:
from tensorflow.keras.models import load_model
import os
import numpy as np
import pandas as pd

In [2]:
model = load_model('Model.h5')

### Scaling the Test images

In [25]:
#defining a function that will scale images
from PIL import Image

def scaling(test_path):
    images = []

    image_path = os.listdir(test_path)
    for x in image_path:
        img = Image.open(test_path + '\\' + x)
        img = img.resize((50,50))
        img = np.array(img)
        images.append(img)

    #Converting images into numpy array
    images = np.array(images)
    #The pixel value of each image ranges between 0 and 255
    #Dividing each image by 255 will scale the values between 0 and 1. This is also known as normalization.
    images = images/255

    return images

**The above function can be used to scale any new traffic-sign images that can be predicted with our model. This is a general purpose function for code reusability.**

In [26]:
dir_path = 'C:\\Users\\Pritam\\Desktop\\Data Science Projets\\Traffic Sign Prediction'
test_path = dir_path + '\\traffic_signs\\Test'

In [27]:
test_images = scaling(test_path)

#### Test labels

In [29]:
test = pd.read_csv(dir_path + '\\traffic_signs\\Test.csv')

In [30]:
y_test = test['ClassId'].values

In [31]:
y_test

array([16,  1, 38, ...,  6,  7, 10], dtype=int64)

######   

### Testing on test images

In [32]:
y_pred = model.predict_classes(test_images)

In [35]:
y_pred

array([16,  1, 38, ..., 32,  7, 10], dtype=int64)

In [33]:
from sklearn.metrics import classification_report

In [36]:
print(classification_report(y_test,y_pred))

              precision    recall  f1-score   support

           0       0.97      1.00      0.98        60
           1       0.98      0.99      0.99       720
           2       0.98      0.99      0.99       750
           3       1.00      0.94      0.97       450
           4       1.00      0.98      0.99       660
           5       0.95      0.98      0.96       630
           6       1.00      0.88      0.94       150
           7       0.99      0.97      0.98       450
           8       0.98      0.98      0.98       450
           9       0.96      1.00      0.98       480
          10       1.00      1.00      1.00       660
          11       0.91      0.99      0.95       420
          12       0.99      0.95      0.97       690
          13       1.00      1.00      1.00       720
          14       1.00      1.00      1.00       270
          15       0.97      1.00      0.98       210
          16       1.00      0.99      1.00       150
          17       1.00    

#### We achieved an overall accuracy of 97% on our model. This is pretty good and we can use this model for predicting some other Traffic signs as well in future.